bulk_grades/api.py

```diff
@@ -437,6 +437,38 @@ def get_rows_to_export(self):
                     row[f'grade-{block_id}'] = effective_grade
             yield row
 
+    def filtered_column_headers(self):
+        """
+        To trim down grade exports, only show subsections which were modified in a bulk update.
+        Returns: a filtered list of columns to export, preserving modified and non-subsection columns
+        """
+        unmodified_subsections = set()
+        for subsection in self._subsections.keys():
+            unmodified_subsections.add(subsection)
+
+        for row in self.result_data:
+            # Ignore rows which didn't introduce changes
+            if row['status'] == 'No Action':
+                continue
+
+            modified_by_row = set()
```

Comment:
nit: a variable named `X_by_Y` tends to reads as a dict of `X`, keyed by `Y`. Maybe just `modified_subsections` or `learner_modified_subsections`?